developer_manual/scripts/read-config.php

```diff
@@ -47,26 +46,27 @@ public function __construct($config = '')
 	}
 
 	/**
-	 * Find the first app directory path that is set as
-	 * being writable and is physically writable in the filesystem
+	 * Find a writable app directory path that is either defined by key 'apps_paths'
+	 * or use the default owncloud_root/apps path if the key is not set 
 	 *
 	 * @return string
 	 * @throws \Exception
 	 */
-	function findPath()
-	{
+	function findPath($ocAppsPath)	{
+
+		// default path = /apps
 		if (!array_key_exists('apps_paths', $this->config)) {
-			throw new \Exception('Configuration is missing the apps_path key');
+			$this->output = $ocAppsPath;
+			return $this->output;
 		}
 
 		foreach ($this->config['apps_paths'] as $path) {
 			if ($path['writable'] == true && is_writable($path['path'])) {
 				$this->output = $path['path'];
-				break;
+				return $this->output;
```

Comment:
(I think) nothing will be returned at this point. I'd rather a default error message, as was there before, instead of providing the user with an empty response, hence the previous default initialisation of `$output`.